ipn/ipnlocal/tailfs.go

```diff
@@ -191,14 +190,16 @@ func (b *LocalBackend) tailfsNotifyShares(shares map[string]*tailfs.Share) {
 
 // tailFSNotifyCurrentSharesLocked sends an ipn.Notify with the current set of
 // TailFS shares.
-func (b *LocalBackend) tailFSNotifyCurrentSharesLocked() {
-	shares, err := b.tailFSGetSharesLocked()
-	if err != nil {
-		b.logf("error notifying current tailfs shares: %v", err)
-		return
-	}
-	// Do the below on a goroutine to avoid deadlocking on b.mu in b.send().
-	go b.tailfsNotifyShares(maps.Clone(shares))
+func (b *LocalBackend) tailFSNotifyCurrentSharesOnceLocked() {
+	b.notifyTailFSSharesOnce.Do(func() {
+		shares, err := b.tailFSGetSharesLocked()
```

Comment:
The most expensive thing on this path is that tailFSGetSharesLocked() unmarshals JSON from the `store`. I had been calling this on every netmap update, but I only need to do it once to send the initial map of shares to the client.